src/common.rs

```diff
@@ -3190,9 +3330,15 @@ impl User {
     }
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub(crate) struct BaseGroupInner {
```

Comment:
Is not possible to declare `GroupInner` here and to implement in the sub modules?